tests/databrowser.js

```diff
@@ -307,25 +309,55 @@ define([
                 });
             });
 
+            describe("with loaded xml", function () {
+                beforeEach(function (done) {
+                    util.init({
+                        plugins: plugins,
+                        javaRosa: {langs: ['en']},
+                        core: {
+                            dataSourcesEndpoint: function (callback) {
+                                event.on("nodeError", function() {
+                                    callback(CASE_DATA);
+                                });
+                            },
+                            form: PRELOADED_HASHTAGS_XML,
+                            onReady: function () {
+                                widget = util.getWidget('property-calculateAttr');
+                                widget.input.promise.then(done);
+                            }
+                        }
+                    });
+                });
+
+                it("should not error for known properties", function() {
+                    assert.strictEqual(widget.getValue(), escapedDobProp);
+                    assert.lengthOf(widget.getControl().find('.label-datanode-unknown'), 0);
+                    event.fire("nodeError");
```

Comment:
picky, and i see you didn't invent this in this PR, but `nodeError` is a weird name if the purpose of the event is to force the external properties to load